### PR TITLE
schedule_package_analyzer

### PR DESCRIPTION
**Pull request by bissenbay**
Created a scheduler for package-analyzer

### CODE REVIEW

File: thoth/common/openshift.py [hunk 1]
```diff
@@ -980,6 +980,80 @@ def get_package_extract_template(self):
 
         return self._get_template("template=package-extract")
 
+    def schedule_package_analyzer(
+        self,
+        package_name: str,
+        package_version: str,
+        index_url: str,
+        output: str,
```
Comment (bissenbay) [inline thoth/common/openshift.py]: Package-extractor had additional parameters such as *, environment_type, origin. Do I need them here as well?

File: thoth/common/openshift.py [hunk 2]
```diff
@@ -980,6 +980,80 @@ def get_package_extract_template(self):
 
         return self._get_template("template=package-extract")
 
+    def schedule_package_analyzer(
+        self,
+        package_name: str,
+        package_version: str,
+        index_url: str,
```
Comment (bissenbay) [inline thoth/common/openshift.py]: Package-extractor had additional parameters such as *, environment_type, origin. Do I need them here as well?

File: thoth/common/openshift.py [hunk 3]
```diff
@@ -980,6 +980,80 @@ def get_package_extract_template(self):
...
+            namespace=self.middletier_namespace,
+            labels={"component": "package-analyzer"},
+        )
+
+    def run_package_analyzer(
+        self,
+        package_name: str,
+        package_version: str,
+        index_url: str,
+        output: str,
```
Comment (bissenbay) [inline thoth/common/openshift.py]: Should I place * parameter to make sure that there might some extra parameters coming? If yes, then where is it good to place?

File: thoth/common/openshift.py (hunk 2 above)
Comment (fridex) [reply thoth/common/openshift.py]: What would these parameters do?

File: thoth/common/openshift.py [hunk 4]
```diff
@@ -980,6 +980,80 @@ def get_package_extract_template(self):
...
+        debug: bool = False,
+        job_id: str = None,
+    ) -> str:
+        """Schedule the given job run, the scheduled job is handled by workload operator based resources available."""
+        if not self.middletier_namespace:
+            raise ConfigurationError(
+                "Unable to schedule package extract job without middletier namespace being set"
```
Comment (fridex) [inline thoth/common/openshift.py]:
```diff
-                "Unable to schedule package extract job without middletier namespace being set"
+                "Unable to schedule package analyzer job without middletier namespace being set"
```

File: thoth/common/openshift.py [hunk 5]
```diff
@@ -980,6 +980,80 @@ def get_package_extract_template(self):
...
+            )
+
+        job_id = job_id or self._generate_id("package-analyzer")
+        parameters = locals()
```
Comment (fridex) [inline thoth/common/openshift.py]: Won't be `job_id` repeated in the run method? (move assignemnt to `parameters` before assignment of to `job_id` environment variable)

File: thoth/common/openshift.py (hunk 3 above)
Comment (fridex) [reply thoth/common/openshift.py]: * is used to delimit keywords arguments from positional arguments. You can place it after `index_url`.

File: thoth/common/openshift.py (hunk 5 above)
Comment (bissenbay) [reply thoth/common/openshift.py]: @fridex I definitely can move ```parameters``` before ```job_id```. However, I just took a look how other schedulers are written and it seems like all of them have ```job_id``` above ```parameters```. WDYT?

Comment (fridex) [reply thoth/common/openshift.py]: Agh, right - you are right. The reference to `job_id` is always present. Keep it as is. Nice catch! :+1:

Comment (bissenbay) [reply thoth/common/openshift.py]: Thanks a lot @fridex !

File: thoth/common/openshift.py [hunk 6]
```diff
@@ -980,6 +980,82 @@ def get_package_extract_template(self):
...
+        if not self.middletier_namespace:
+            raise ConfigurationError(
+                "Running package-analyzer requires middletier namespace to be specified"
+            )
+
+        template = template or self.get_package_analyzer_template()
+
+        self.set_template_parameters(
+            template,
+            THOTH_PACKAGE_ANALYZER_PACKAGE_NAME=package_name,
```
Comment (bissenbay) [inline thoth/common/openshift.py]: I changed environment variables on [this PR](https://github.com/thoth-station/package-analyzer/pull/4/files) to ```package-analyzer``` as well.

**Pull request merged**